src/frequenz/dispatch/actor.py

```diff
@@ -243,9 +258,11 @@ async def _update_dispatch_schedule_and_notify(
     def _update_timer(self) -> None:
         """Update the timer to the next event."""
         if self._scheduled_events:
-            due_at: datetime = self._scheduled_events[0][0]
+            due_at: datetime = self._scheduled_events[0].priority[0]
```

Comment:
If the `priority` is not opaque maybe it makes more sense to have 2 different fields in `QueueItem`:

```diff
-priority: tuple[datetime, int]
+time: datetime
+dispatch_id: int
```